syntax = "proto2";
message RequestEnvelop {
  required int32 unknown1 = 1;
  optional int64 rpc_id = 3;
  repeated Requests requests = 4;
  optional Unknown6 unknown6 = 6;
  optional fixed64 latitude = 7;
  optional fixed64 longitude = 8;
  optional fixed64 altitude = 9;
  optional AuthInfo auth = 10;
  optional int64 unknown12 = 12;

  message Requests {
    required int32 type = 1;
    optional bytes message=2;
  }

  message four2int{
      optional uint64 value=1;
  }
   
  message four2string{
      optional string value=1;
      optional fixed64 PlayerLatDegrees = 2;
      optional fixed64 PlayerLngDegrees = 3;
      optional fixed64 GymLatDegrees = 4;
      optional fixed64 GymLngDegrees = 5;
  }

  message Unknown6 {
    required int32 unknown1 = 1;
    required Unknown2 unknown2 = 2;

    message Unknown2 {
      required bytes unknown1 = 1;
    }

  }
  message AuthInfo {
    required string provider = 1;
    required JWT token = 2;

    message JWT {
      required string contents = 1;
      required int32 unknown13  = 2;
    }
  }
}

message ResponseEnvelop {
  required int32 unknown1 = 1;
  optional int64 unknown2 = 2;
  optional string api_url = 3;
  optional Unknown6 unknown6 = 6;
  optional Unknown7 unknown7 = 7;
  repeated Payload payload = 100;

  message Unknown6 {
    required int32 unknown1 = 1;
    required Unknown2 unknown2 = 2;

    message Unknown2 {
      required bytes unknown1 = 1;
    }

  }

  message Unknown7 {
    optional bytes unknown71 = 1;
    optional int64 unknown72 = 2;
    optional bytes unknown73 = 3;
  }

  message Payload {
    optional Gym value=1;
  }

  message four2int{
      optional uint64 value=1;
  }
   
  message four2string{
      optional Gym value=1;
  }

  message Gym {
      optional GymInfo gym = 1;
      repeated GymPlayers players = 2; 

      message GymInfo {
        optional string GymState = 1;
        optional string Name = 2;
        optional string Url = 3;
        optional int32 Result = 4;
        optional string Description = 5;
      }

      message GymPlayers {
        optional Pokemon pokemon = 1;
        optional Player2 owner = 2;

        message Pokemon {
          optional fixed64 Id = 1;
          optional int32 PokemonId = 2;
          optional int32 Cp = 3;
          optional int32 Stamina = 4;
          optional int32 MaxStamina = 5;
          optional int32 Move1 = 6;
          optional int32 Move2 = 7;
          optional int32 DeployedFortId = 8;
          optional string OwnerName = 9;
          optional int32 IsEgg = 10;
          optional int32 EggKmWalkedTarget = 11;
          optional int32 EggKmWalkedStart = 12;
          optional int32 Origin = 14;
          optional fixed64 HeightM = 15;
          optional fixed64 WeightKg = 16;
          optional int32 IndividualAttack = 17;
          optional int32 IndividualDefense = 18;
          optional int32 IndividualStamina = 19;
          optional fixed64 CpMultiplier = 20;
          optional int32 Pokeball = 21;
          optional int32 CapturedS2CellId = 22;
          optional int32 BattlesAttacked = 23;
          optional int32 BattlesDefended = 24;
          optional int32 EggIncubatorId = 25;
          optional int32 CreationTimeMs = 26;
          optional int32 NumUpgrades = 27;
          optional fixed64 AdditionalCpMultiplier = 28;
          optional int32 Favorite = 29;
          optional int32 Nickname = 30;
          optional int32 FromFort = 31;
        }
        message Player2 {
          optional string Name = 1;
          optional int32 Level = 2;
          optional Avatar Avatar = 3;
        }
        message Avatar {
          optional int32 Avatar = 8;
          optional int32 Skin = 2;
          optional int32 Hair = 3;
          optional int32 Shirt = 4;
          optional int32 Pants = 5;
          optional int32 Hat = 6;
          optional int32 Shoes = 7;
          optional int32 Eyes = 9;
          optional int32 Backpack = 10;
        }
      }
    }

}
